File: backend/Data/Entities/Auction.cs
namespace Ride.Api.Data.Entities;

public class Auction
{
    public Guid Id { get; set; }
    public Guid CarId { get; set; }
    public Car Car { get; set; } = null!;

    public Guid OwnerId { get; set; }
    public ApplicationUser Owner { get; set; } = null!;

    public decimal StartPriceEur { get; set; }
    public decimal? BuyNowPriceEur { get; set; }
    public DateTime StartsAt { get; set; }
    public DateTime EndsAt { get; set; }
    public bool IsActive { get; set; } = true;
    public string? Description { get; set; }

    public List<string> Issues { get; set; } = new();
    public List<string> ImageUrls { get; set; } = new();
    public List<string> VideoUrls { get; set; } = new();

    public ICollection<AuctionBid> Bids { get; set; } = new List<AuctionBid>();

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: src/models/Auction.ts
import { Car } from './Car';

export interface AuctionBid {
  id: string;
  bidderName: string;
  amountEur: number;
  createdAt: string;
}

export interface Auction {
  id: string;
  carId: string;
  car: Car;
  startPriceEur: number;
  buyNowPriceEur?: number | null;
  currentPriceEur: number;
  startsAt: string;
  endsAt: string;
  isActive: boolean;
  hasStarted: boolean;
  hasEnded: boolean;
  issues: string[];
  imageUrls: string[];
  videoUrls: string[];
  bids: AuctionBid[];
}
